authenticator/authenticator.go

```diff
@@ -156,30 +291,25 @@ func (a *Authenticator) GetPGSHA256Hash(ctx context.Context, req *pb.PGSHA256Has
 		return nil, status.Errorf(codes.InvalidArgument, err.Error())
 	}
 
-	salt := req.GetSalt()
-
-	if len(salt) == 0 {
-		msg := fmt.Sprintf("salt not provided")
-		log.Error(msg)
-		return nil, status.Errorf(codes.InvalidArgument, msg)
-	}
-	iterations := int(req.GetIterations())
-	saltedPass, err := computePGSHA256SaltedPass(password, salt, iterations)
+	saltedPass, err := computePGSHA256SaltedPass(password, salt, int(iterations))
 	if err != nil {
 		msg := fmt.Sprintf("Could not compute hash %s", err)
 		log.Error(msg)
 		return nil, status.Errorf(codes.InvalidArgument, msg)
 	}
 
-	authMsg := req.GetAuthenticationMsg()
-
-	if len(authMsg) == 0 {
-		msg := fmt.Sprintf("authentication message not provided")
-		log.Error(msg)
-		return nil, status.Errorf(codes.InvalidArgument, msg)
-	}
 	cproof := computePGSHA256Cproof(saltedPass, authMsg)
 	sproof := computePGSHA256Sproof(saltedPass, authMsg)
+
+	// Make sure the ARN they claimed they had to get the creds was their actual ARN.
+	select {
+	case verifiedIAMArn := <-verifiedIAMArnChan:
+		if verifiedIAMArn != claimedIamArn {
```

Comment:
TODO - test if it would be safe to allow a role arn to match an assumed role arn.

Reply:
Pushed in https://github.com/approzium/approzium/pull/47/commits/427466dda7cf7c1b3c451e9d440dc34cfd9e0a5d